### PR TITLE
chore: Added bothrs eslint-config but broke a few things

### PR DESCRIPTION
**Pull request by Vanatis**
## What
Added Bothrs/eslint-config to the monorepo.
Added eslinting to all subrepos
Fixed linting issues in subrepos

## Ticket
R&D task

## Analytics

- [ ] Added analytics
- [ ] Documented analytics

### CODE REVIEW

File: packages/react-native-layout/package.json [hunk 1]
```diff
@@ -7,20 +7,6 @@
...
-    "ios",
-    "cpp",
-    "bothrs-layout.podspec",
-    "!lib/typescript/example",
-    "!android/build",
-    "!ios/build",
-    "!**/__tests__",
-    "!**/__fixtures__",
-    "!**/__mocks__"
-  ],
```
Comment (thgh) [inline packages/react-native-layout/package.json:23]: Why would you remove this?

File: packages/react-native-layout/src/components/spacing/Spacing.tsx [hunk 2]
```diff
@@ -1,10 +1,9 @@
 import React from 'react'
 import { View } from 'react-native'
-import type { ViewStyle } from 'react-native'
 
 import type { TestableComponent } from '../../types/generic-props'
-
 import type { SpacingProps } from './Spacing.props'
+import type { ViewStyle } from 'react-native'
```
Comment (thgh) [inline packages/react-native-layout/src/components/spacing/Spacing.tsx:6]: Got import-sort working?

Comment (Vanatis) [reply packages/react-native-layout/src/components/spacing/Spacing.tsx:6]: Yup!

File: packages/react-native-layout/package.json (hunk 1 above)
Comment (Vanatis) [reply packages/react-native-layout/package.json:23]: This is a draft where i was testing a ton of things out in order to make eslint work. Got it figured out now and this'll be fixed in the next commit / actual PR :)

File: packages/react-native-layout/.eslintrc.js [hunk 3]
```diff
@@ -0,0 +1,15 @@
+module.exports = { 
+  extends: [
+    '../../.eslintrc.js',
+    '@bothrs/eslint-config/react',
+    '@bothrs/eslint-config/react-native',
+  ],
+  parserOptions: {
+    project: 'tsconfig.json',
+    tsconfigRootDir: __dirname,
```
Comment (bramvanhoutte) [inline packages/react-native-layout/.eslintrc.js:9]: Q: Why is this necessary? Normally specifying the project field as `./tsconfig.json` is enough

Comment (Vanatis) [reply packages/react-native-layout/.eslintrc.js:9]: It isn't enough. The bug that I've showed before where either the formatting **or** the yarn lint is working still happened. This is due to the way that yarn workspaces works. When developing and changing files, everything assumes you start from the root. Once you run the yarn lint command on root however, it runs a yarn lint command from each package separately, which means their point of origin is different.

Adding the __dirname, fixes this by always working from the same origin point for both situations

**Comment by FabianMeul** [comment]
LGTM

**Review by FabianMeul — APPROVED** [review]
